go/vt/wrangler/reparent.go

```diff
@@ -490,11 +487,19 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 		if alias == masterElectTabletAliasStr {
 			continue
 		}
+		if alias == topoproto.TabletAliasString(oldMasterTabletInfo.Alias) {
+			// PromoteSlaveWhenCaughtUp sets tabletType on newMaster to MASTER
+			// that immediately triggers an update of the shard record
+			// which in turn notifies the old master to demote itself to REPLICA
+			// so we don't need to do anything here
+			continue
+		}
 		wgReplicas.Add(1)
 		go func(alias string, tabletInfo *topo.TabletInfo) {
 			defer wgReplicas.Done()
 			wr.logger.Infof("setting new master on slave %v", alias)
 
+			// TODO(deepthi): review and fix this comment
```

Comment:
This comment was meant to express the principle mentioned above: we need to be able to do everything without knowing in advance who the "old master" was.

Do you have a concern about keeping `forceStartReplication` as `false` long-term?

Reply:
Only that with the current set of changes in this PR, it needed to be reviewed (because we are excluding oldMaster from calls to SetMaster). The point is moot now that we have decided to make SetMaster idempotent.